src/jshint.js

```diff
@@ -2743,6 +2743,9 @@ var JSHINT = (function() {
           break;
         }
       } else {
+        if (state.option.trailingcomma && state.inES5()) {
```

Comment:
JSHint has a few IE-specific warnings (e.g. `W070`, "Extra comma. (it breaks
older versions of IE)"), and rather than referencing an option dedicated to
that condition, it applies these a little inconsistently--when the environment
does not support ES5. So this behavior respects precedence.

But like you, I'm not sure the precedent is a good one. I'm thinking that a new
"enforcing" option like `oldie` might be good, especially from a language
purist's perspective. I've written up a proposal here: gh-3094